sdk/eventgrid/azure-eventgrid/azure/eventgrid/_consumer.py

```diff
@@ -0,0 +1,50 @@
+# coding=utf-8
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is regenerated.
+# --------------------------------------------------------------------------
+
+from typing import TYPE_CHECKING
+import json
+import six
+import logging
+
+from azure.core import PipelineClient
+from msrest import Deserializer, Serializer
+
+if TYPE_CHECKING:
+    # pylint: disable=unused-import,ungrouped-imports
+    from typing import Any
+
+_LOGGER = logging.getLogger(__name__)
+
+from ._models import DeserializedEvent
+
+class EventGridConsumer(object):
+    """
+    A consumer responsible for deserializing event handler messages, to allow for access to strongly typed Event objects.
+    """
+
+    def deserialize_event(self, event, **kwargs):
```

Comment:
Given the direction conversations are going ( per discussions [here](https://gist.github.com/kerri-lee/900f7fcee8ba1bcca5ee18e19e9b59d3) ) it's looking like the shape of this API may be changing. Items such as:
- location, here or on the event models
  - With this, not utilizing a wrapper class (deserializedEvent) and instead doing the "lazy load" when accessing the data property.
- parsing directly from e.g. a requestContext

 We're going to need to have a longer conversation when @lmazuel is back.  I'm not too concerned if this is the base we work off of since this logic will presumably form the core regardless, especially since the discussion is still in flight, but was curious to hear your thoughts (@rakshith91), largely on whether my instinct to iterate from working to working since we're preview isn't disastrous, or if you'd wait to hit master until we locked fully.

Reply:
LEt this not block us since we haven't settled on design yet - have an issue tracking it
https://github.com/Azure/azure-sdk-for-python/issues/13227